Description: Avoid a potential nullptr dereference
Origin: upstream, https://lab.louiz.org/louiz/biboumi/commit/eb8f1cb
Author: louiz’ <louiz@louiz.org>
Forwarded: yes
Last-Update: 2016-12-21

--- a/src/xmpp/biboumi_component.cpp
+++ b/src/xmpp/biboumi_component.cpp
@@ -570,13 +570,11 @@
     Jid to(stanza.get_tag("to"));
 
     const XmlNode* query = stanza.get_child("query", MAM_NS);
-    std::string query_id;
-    if (query)
-      query_id = query->get_tag("queryid");
 
     Iid iid(to.local, {'#', '&'});
-    if (iid.type == Iid::Type::Channel && to.resource.empty())
+    if (query && iid.type == Iid::Type::Channel && to.resource.empty())
       {
+        const std::string query_id = query->get_tag("queryid");
         std::string start;
         std::string end;
         const XmlNode* x = query->get_child("x", DATAFORM_NS);
