Description: Avoid cmake messing with git
Author: Jonas Smedegaard <dr@jones.dk>
Last-Update: 2016-12-21

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -198,23 +198,6 @@
   database)
 endif()
 
-include(ExternalProject)
-ExternalProject_Add(catch
-  GIT_REPOSITORY "https://lab.louiz.org/louiz/Catch.git"
-  PREFIX "external"
-  UPDATE_COMMAND ""
-  CONFIGURE_COMMAND ""
-  BUILD_COMMAND ""
-  INSTALL_COMMAND ""
-  )
-set_target_properties(catch PROPERTIES EXCLUDE_FROM_ALL TRUE)
-ExternalProject_Get_Property(catch SOURCE_DIR)
-if(NOT EXISTS ${CMAKE_SOURCE_DIR}/tests/catch.hpp)
-  target_include_directories(test_suite
-    PUBLIC "${SOURCE_DIR}/include/"
-    )
-  add_dependencies(test_suite catch)
-endif()
 add_custom_target(check COMMAND "test_suite" "-s"
   DEPENDS test_suite biboumi)
 add_custom_target(e2e COMMAND "python3" "${CMAKE_CURRENT_SOURCE_DIR}/tests/end_to_end/"
@@ -258,7 +241,6 @@
   COMMAND ${CMAKE_COMMAND} -E cmake_echo_color --cyan "${ARCHIVE_NAME}.tar.xz created."
   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
   )
-add_dependencies(dist catch)
 
 add_custom_target(rpm
   COMMAND mkdir -p rpmbuild/{BUILD,RPMS,SOURCES,SPECS,SRPMS}
